src/VirtoCommerce.InventoryModule.Data/Services/FulfillmentCenterService.cs

```diff
@@ -37,20 +37,21 @@ public virtual async Task<IEnumerable<FulfillmentCenter>> GetByIdsAsync(IEnumera
                 throw new ArgumentNullException(nameof(ids));
             }
             var cacheKey = CacheKey.With(GetType(), nameof(GetByIdsAsync), string.Join("-", ids.OrderBy(x => x)));
-            return await _platformMemoryCache.GetOrCreateExclusiveAsync(cacheKey, async (cacheEntry) =>
+            var result = await _platformMemoryCache.GetOrCreateExclusiveAsync(cacheKey, async (cacheEntry) =>
             {
+                using var repository = _repositoryFactory();
+                //It is so important to generate change tokens for all ids even for not existing objects to prevent an issue
```

Comment:
I think you should remove these changes or discuss with @kostyrin, because they might conflict with changes related to this ticket https://virtocommerce.atlassian.net/browse/VP-2046
moreover, you use for caching the entire region and do not use passed ids to build keys for the cache and this comment hasn't any sense.

Reply:
I'm agree. Do it later. Key change there is **Clone**.